apis/kueue/v1beta1/workload_types.go

```diff
@@ -382,6 +393,21 @@ type ReclaimablePod struct {
 	Count int32 `json:"count"`
 }
 
+type PodSetRequest struct {
+	// name is the name of the podSet. It should match one of the names in .spec.podSets.
+	// +kubebuilder:default=main
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern="^(?i)[a-z0-9]([-a-z0-9]*[a-z0-9])?$"
+	Name string `json:"name"`
+
+	// resources is the total resources all the pods in the podset need to run.
+	//
+	// Beside what is provided in podSet's specs, this value also takes into account
+	// the LimitRange defaults and RuntimeClass overheads at the moment of consideration
+	// and the application of resource.excludeResourcePrefixes and resource.transformations.
+	Resources corev1.ResourceList `json:"resources,omitempty"`
```

Comment:
This is just a comment for better understanding.
The resources field should not make pointer as you defined here since the corev1.ResourceList is typed map, and the map is already pointer.